code/missionui/fictionviewer.cpp

```diff
@@ -560,55 +526,46 @@ void fiction_viewer_reset()
 	}
 }
 
-void fiction_viewer_load(const char *background_640, const char *background_1024, int ui_index, const char *filename, const char *font_filename, const char *voice_filename)
+void fiction_viewer_load(int stage)
 {
-	int file_length;
-	Assertion(filename, "Invalid fictionviewer filename pointer given!");
-	Assertion(font_filename, "Invalid fictionviewer font filename pointer given!");
-	Assertion(voice_filename, "Invalid fictionviewer voice filename pointer given!");
+	Assertion(stage >= 0 && static_cast<size_t>(stage) < Fiction_viewer_stages.size(), "stage parameter must be in range of Fiction_viewer_stages!");
 
 	// just to be sure
 	if (Fiction_viewer_text != NULL)
 	{
-		Int3();
-		fiction_viewer_reset();
+		Error(LOCATION, "Fiction viewer text should be NULL here!  Trace out and fix!");
```

Comment:
It has a fixed-size buffer, but it uses vsnprintf(), so if the string is too long it'll just truncate it.